openstack/networking/v2/extensions/provider/results.go

```diff
@@ -48,6 +51,29 @@ type NetworkExtAttrs struct {
 	SegmentationID string `json:"provider:segmentation_id"`
 }
 
+func (n *NetworkExtAttrs) UnmarshalJSON(b []byte) error {
+	type tmp NetworkExtAttrs
+	var networkExtAttrs *struct {
+		tmp
+		SegmentationID interface{} `json:"provider:segmentation_id"`
+	}
+
+	if err := json.Unmarshal(b, &networkExtAttrs); err != nil {
+		return err
+	}
+
+	*n = NetworkExtAttrs(networkExtAttrs.tmp)
+
+	switch t := networkExtAttrs.SegmentationID.(type) {
+	case float64:
+		n.SegmentationID = strconv.FormatFloat(t, 'f', -1, 64)
+	case string:
+		n.SegmentationID = string(t)
```

Comment:
Nothing wrong with this, but because we know `t` is of type `string` here (because we're in `case string`), we should be able to just:
```go
n.SegmentationID = t
``` 
(Note: no change required, more of a for-future-reference comment)